test/gpu/native/studies/sort/radixSortGpu.chpl

```diff
@@ -15,101 +16,198 @@ module RadixSort {
   // How many bits we sort at once based on the bitsAtATime
   const buckets = 1 << bitsAtATime; // 2^bitsAtATime, ex: 2^4 = 16 = 0b10000
   const bitMask = buckets - 1; // 2^bitsAtATime - 1, ex: 2^4 - 1 = 15 = 0b1111
-
-  proc serialScan(ref prefixSum: [] uint, buckets: int){
+  // The size of the chunks the input array is divided into for parallelism
+  // Each chunk has it's own count table (hence it's own prefix sum)
+  // Which we then intelligently combine together to produce a sorted output
+  config const chunkSize = 128;
+  config const scanChunkSize = 1024;
+  // TODO Choose better defaults for the above two things
+  config const parallelScan = false; // Need to improve this more
+
+  proc serialScan(ref arr: [] uint, in startIdx = arr.domain.low, in endIdx = arr.domain.high){
```

Comment:
Drop `in`s

Reply:
The default intent seems to be `const` and I modify the `startIdx` and `endIdx` in case they're outside the bounds of the array.
That's why I added the `in` intents.

Reply:
Hmm.. In terms of code structure, this feels wrong to me. Alternatives in order of my preference:

1. default intent here, optionally add bounds checking logic `if boundsCheck` (that'll be true without `--fast`, false otherwise), adjust the callsite of this function to ensure that it passes correct bounds, document it as a short and sweet comment at the top of this function
2. default intent here, document that it is valid to pass values that are OOB, inside the function declare local variables that are adjusted to correct bounds instead of reusing arguments

what do you think?

Reply:
I think I can do the first option here